fluture.js

```diff
@@ -132,6 +131,43 @@
     return function partial(d){ return f(a, b, c, d) };
   }
 
+  //Creates a dispatcher for a nullary method.
+  function createNullaryDispatcher(method){
+    return function nullaryDispatch(m){
+      if(m && typeof m[method] === 'function') return m[method]();
+      error$invalidArgument(`Future.${method}`, 1, `have a "${method}" method`, m);
+    };
+  }
+
+  //Creates a dispatcher for a unary method.
+  function createUnaryDispatcher(method){
+    return function unaryDispatch(a, m){
+      if(arguments.length === 1) return unaryPartial(unaryDispatch, a);
+      if(m && typeof m[method] === 'function') return m[method](a);
+      error$invalidArgument(`Future.${method}`, 1, `have a "${method}" method`, m);
+    };
+  }
+
+  //Creates a dispatcher for a binary method.
+  function createBinaryDispatcher(method){
+    return function binaryDispatch(a, b, m){
+      if(arguments.length === 1) return unaryPartial(binaryDispatch, a);
+      if(arguments.length === 2) return binaryPartial(binaryDispatch, a, b);
+      if(m && typeof m[method] === 'function') return m[method](a, b);
+      error$invalidArgument(`Future.${method}`, 2, `have a "${method}" method`, m);
+    };
+  }
+
+  //Creates a dispatcher for a binary method, but takes the object first rather than last.
+  function createInvertedBinaryDispatcher(method){
+    return function invertedBinaryDispatch(m, a, b){
+      if(arguments.length === 1) return unaryPartial(invertedBinaryDispatch, m);
+      if(arguments.length === 2) return binaryPartial(invertedBinaryDispatch, m, a);
+      if(m && typeof m[method] === 'function') return m[method](a, b);
+      error$invalidArgument(`Future.${method}`, 0, `have a "${method}" method`, m);
+    };
+  }
+
```

Comment:
These will eventually disappear as I will depend on `sanctuary-type-classes` for FantasyLand dispatching, rewrite all other dispatchers to just construct their associated classes directly.